### PR TITLE
[IOTDB-1193] Remove redundant sync meta leader in CManager for cluster module

### PR DESCRIPTION
**Pull request by LebronAl**
see [1193](https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-1193?filter=allopenissues)

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java [hunk 1]
```diff
@@ -1335,12 +1332,6 @@ private boolean tryPullTimeSeriesSchemas(Node node, PullSchemaRequest request) {
    * @return all paths after removing wildcards in the path
    */
   public List<PartialPath> getMatchedPaths(PartialPath originPath) throws MetadataException {
-    // make sure this node knows all storage groups
-    try {
-      metaGroupMember.syncLeaderWithConsistencyCheck(false);
-    } catch (CheckConsistencyException e) {
-      throw new MetadataException(e);
-    }
```
Comment (neuyilan) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:1343]: Hi, @LebronAl I see there are so many places use this method, how to make sure that this modification has no effect on the other logic that calls this function?

The same as other methods.

Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:1343]: Manually review their call stacks one by one to ensure that a query request only performs synchronization once.

Maybe we need to analyse the current synchronization of CMManager and fix it further.

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by mychaow — APPROVED** [review]
LGTM

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 1 above)
Comment (neuyilan) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:1343]: Ok

**Pull request merged**